force-app/main/default/classes/PS_CommitmentRequest.cls

```diff
@@ -194,6 +194,17 @@ public inherited sharing class PS_CommitmentRequest {
             .withBody(jsonRequestBody)
             .build();
     }
+    public static HttpRequest buildRequest(String commitmentId, String jsonRequestBody,
```

Comment:
The above version of `buildRequest` could call this one after determining the `method`, avoids repetition.